presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveUtil.java

```diff
@@ -308,11 +309,14 @@ private static void configureCompressionCodecs(JobConf jobConf)
 
             Class<? extends InputFormat<?, ?>> inputFormatClass = getInputFormatClass(jobConf, inputFormatName);
             if (symlinkTarget && inputFormatClass == SymlinkTextInputFormat.class) {
-                // Symlink targets are assumed to be TEXTFILE unless serde indicates otherwise.
-                inputFormatClass = TextInputFormat.class;
-                if (isDeserializerClass(schema, AvroSerDe.class)) {
-                    inputFormatClass = AvroContainerInputFormat.class;
+                String serDe = getDeserializerClassName(schema);
+                for (HiveStorageFormat format : EnumSet.allOf(HiveStorageFormat.class)) {
+                    if (serDe.equals(format.getSerDe())) {
+                        inputFormatClass = HiveUtil.getInputFormatClass(jobConf, format.getInputFormat());
+                        return ReflectionUtils.newInstance(inputFormatClass, jobConf);
+                    }
                 }
+                throw new PrestoException(INVALID_TABLE_PROPERTY, format("No corresponding InputFormat for SerDe %s for SymlinkTextInputFormat table", serDe));
```

Comment:
`INVALID_TABLE_PROPERTY` is for when the user is creating a table in Presto and specifies an invalid table property. We have `HIVE_UNSUPPORTED_FORMAT` which seems to exactly match this error case. Also, it's better to put the variable part of the error message at the end, so that the fixed part of the message is more searchable.

How about
```java
throw new PrestoException(HIVE_UNSUPPORTED_FORMAT, "Unknown SerDe for SymlinkTextInputFormat: " + serDe);